include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -19,259 +20,178 @@
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_output_format(sequence_file_output_format const &)                      = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_output_format & operator=(sequence_file_output_format const &)          = delete;
+    sequence_file_output_format(sequence_file_output_format &&)                  noexcept = default; //!< Defaulted.
+    sequence_file_output_format & operator=(sequence_file_output_format &&)      noexcept = default; //!< Defaulted.
+    ~sequence_file_output_format()                                               noexcept = default; //!< Defaulted.
+    //!\}
 
     //!\copydoc SequenceFileOutputFormat::write
     template <typename stream_type,     // constraints checked by file
               typename seq_type,        // other constraints checked inside function
               typename id_type,
               typename qual_type>
     void write(stream_type                        & stream,
-               sequence_file_output_options const & options,
+               sequence_file_output_options const & SEQAN3_DOXYGEN_ONLY(options),
                seq_type                           && sequence,
                id_type                            && id,
                qual_type                          && qualities)
     {
-        if constexpr (!(detail::decays_to_ignore_v<seq_type>))
-        {
-            static_assert(std::ranges::ForwardRange<seq_type> && Alphabet<reference_t<seq_type>>,
-                          "The sequence must model std::ranges::ForwardRange and its value type must model "
-                          "seqan3::Alphabet.");
-        }
-        seqan3::ostreambuf_iterator stream_it{stream};
-        // ID
-        if constexpr (detail::decays_to_ignore_v<id_type>)
-            stream_it = '*';
-        else if (std::ranges::empty(id)) //[[unlikely]]
-            stream_it = '*';
-        else
-            std::ranges::copy(id, stream_it);
+        using default_align_t = std::pair<std::span<gapped<char>>, std::span<gapped<char>>>;
+        using default_mate_t  = std::tuple<std::string_view, std::optional<int32_t>, int32_t>;
 
-        stream << "\t0\t*\t0\t0\t*\t*\t0\t0\t";
+        alignment_file_output_options output_options;
 
-        // Sequence
-        if constexpr (detail::decays_to_ignore_v<seq_type>)
-            stream_it = '*';
-        else if (std::ranges::empty(sequence)) //[[unlikely]]
-            stream_it = '*';
-        else
-            std::ranges::copy(sequence | view::to_char, stream_it);
-        stream_it = '\t';
+        align_format.write(stream, output_options, std::ignore,
+                           default_or(sequence), default_or(qualities), default_or(id),
+                           0, std::string{}, std::string{}, -1, default_align_t{}, 0, 0,
```

Comment:
```suggestion
                           0, std::string_view{}, std::string_view{}, -1, default_align_t{}, 0, 0,
```
:nail_care: empty string views are guaranteed to be created without allocation while std::strings might allocate.